src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -22,6 +22,16 @@ def _resolve_aliases(target):
     else:
       yield declared
 
+    for export in _resolve_exports(declared):
+      yield export
+
+
+def _resolve_exports(target):
+  for export in getattr(target, 'exports', []):
+    for exp in _resolve_exports(export):
```

Comment:
I think it ought to work like this: A has C on its compile classpath. Given an X depending on A, X won't have C on its compile classpath because A doesn't export B or C. If A exports B, then C will be on X's classpath.

```
A -d-> B
B -e-> C
X -d-> A

cp(A) == {B, C}
cp(X) == {A}

# if we add
A -e-> B
# then
cp(X) == {A, B, C}
```